source/features/user-local-time.tsx

```diff
@@ -80,9 +80,9 @@ function parseOffset(date: string): number {
 function init(): void {
 	const hovercard = select('.js-hovercard-content > .Popover-message')!;
 
-	observeElement(hovercard, async () => {
+	observeElement(hovercard, async ([mutation]) => {
 		if (
-			select.exists('.rgh-local-user-time', hovercard) || // Time already added
+			mutation?.addedNodes?.length !== 1 || // Avoid duplicated time element issue #3634
```

Comment:
This condition is too broad. There are a number of situations where this condition could be false and repeated. 

I can think of 2 ways to solve this issue, either
- mark the container with a class as soon as the API call starts, or 
- use `mem()` so that it deduplicates call to the container (this is too hard to explain, but we have other examples of deduplication via mem)


In both cases, the “container” is what we select in: `select('div.d-flex.mt-3 > div.overflow-hidden.ml-3', hovercard)!.append(container)`

Reply:
But actually we should just use observeSelector probably 🤔
We already know how to deduplicate that

Reply:
> We already know how to deduplicate that

You mean adding a class and doing `:not(.this-class)` in the selector? I guess that could work if we add the class at the very beginning of the mutation handler.

Reply:
Yes. Keep in mind that the handler will need to fire multiple times on the same hovercard element since only its content is updated